dulwich/index.py

```diff
@@ -604,6 +604,32 @@ def read_submodule_head(path):
         return None
 
 
+def has_directory_changed(error, tree_path, entry):
+    """ When handling an error trying to create a blob from a path, call this
```

Comment:
Please start docstrings with a single line summarizing what the function does, for e.g. sphinx.